eos-payg-ctl/eos-payg-ctl

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+#
+# Copyright © 2018 Endless Mobile, Inc.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+#
+# Alternatively, the contents of this file may be used under the terms of the
+# GNU Lesser General Public License Version 2.1 or later (the "LGPL"), in
+# which case the provisions of the LGPL are applicable instead of those above.
+# If you wish to allow use of your version of this file only under the terms
+# of the LGPL, and not to allow others to use your version of this file under
+# the terms of the MPL, indicate your decision by deleting the provisions
+# above and replace them with the notice and other provisions required by the
+# LGPL. If you do not delete the provisions above, a recipient may use your
+# version of this file under the terms of either the MPL or the LGPL.
+
+import argparse
+import contextlib
+import datetime as dt
+import sys
+from dateutil import tz
+from gi.repository import GLib, Gio
+
+BUS_NAME = 'com.endlessm.Payg1'
+OBJECT_PATH = '/com/endlessm/Payg1'
+INTERFACE = 'com.endlessm.Payg1'
+ERROR_DOMAIN = 'com.endlessm.Payg1.Error'
+
+DBUS_PROPERTIES_INTERFACE = 'org.freedesktop.DBus.Properties'
+
+
+def __format_utc_timestamp(timestamp):
+    expiry_utc = dt.datetime.fromtimestamp(timestamp, tz=tz.tzutc())
+    expiry_local = expiry_utc.astimezone(tz.tzlocal())
+    return str(expiry_local)
+
+
+def __get_proxy():
+    return Gio.DBusProxy.new_for_bus_sync(
+        Gio.BusType.SYSTEM,
+        Gio.DBusProxyFlags.NONE,
+        None,
+        BUS_NAME,
+        OBJECT_PATH,
+        INTERFACE,
+        None,  # cancellable
+    )
+
+
+def command_status(proxy):
+    '''Show the current PAYG status of this device.'''
+    props = {
+        name: proxy.get_cached_property(name).unpack()
+        for name in proxy.get_cached_property_names()
+    }
+
+    width = max(map(len, props), 0)
+    formatters = {
+        'ExpiryTime': __format_utc_timestamp,
+        'RateLimitEndTime': __format_utc_timestamp,
+    }
+
+    for key, value in sorted(props.items()):
+        try:
+            formatter = formatters[key]
+        except KeyError:
+            formatted = value
+        else:
+            formatted = formatter(value)
```

Comment:
I almost wrote this as

```python
formatted = formatters.get(key, lambda x: x)(value)
```

and actually I think that's clearer now I type it out…